### PR TITLE
returns sorted list of ssh keys from the API

### LINKED ISSUE #1422: Flaky TestListSSHKeysAssignedToClusterEndpoint
child of #966

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: returns sorted list of ssh keys from the API

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1422 and #1432 

Release note:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/provider/kubernetes/sorter.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/integer"
+	"k8s.io/client-go/util/jsonpath"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+func sortObjects(decoder runtime.Decoder, objs []runtime.Object, fieldInput string) (*runtimeSort, error) {
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/sorter.go:19]: shamelessly copied from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/sorter.go :)

File: api/pkg/handler/cluster_test.go [hunk 2]
```diff
@@ -15,11 +15,20 @@ import (
 	kubermaticv1 "github.com/kubermatic/kubermatic/api/pkg/crd/kubermatic/v1"
 	"github.com/kubermatic/kubermatic/api/pkg/kubernetes"
 
+	"time"
```
Comment (mrIncompetent) [inline api/pkg/handler/cluster_test.go]: grouping

File: api/pkg/handler/cluster_test.go [hunk 3]
```diff
@@ -15,11 +15,20 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 func TestListSSHKeysAssignedToClusterEndpoint(t *testing.T) {
+	const longForm = "Jan 2, 2006 at 3:04pm (MST)"
+	creationTime, err := time.Parse(longForm, "Feb 3, 2013 at 7:54pm (PST)")
+	if err != nil {
+		t.Fatal(err)
+
```
Comment (mrIncompetent) [inline api/pkg/handler/cluster_test.go]: remove empty line

**Comment by mrIncompetent** [comment]
Implementation is fine.
But are we sure we want this? 
I guess we must now also implement sorting for every other endpoint which returns a list?

Sorting itself is nice, but our only consumer - the dashboard - already sorts the response.
My concern is that we'll add quite some code just to be able to sort the response for the tests.

**Comment by p0lyn0mial** [comment]
I took it for granted. Primarily because as you already pointed out, it is nice to return stable responses to the clients. Introducing sorting by the creation date was a step into that direction. On the other hand the API dosn't have to make any assumptions about how the client wants to present the result to the users. 

I'm willing to change/remove the sorting, what's your opinion ?

**Comment by mrIncompetent** [comment]
Fine for now. I just want to prevent that we turn this into a api feature

**Review by mrIncompetent — DISMISSED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Review by kgroschoff — APPROVED** [review]
LGTM

**Pull request merged**